### PR TITLE
feat: report provisioning info

### PR DESCRIPTION
**Pull request by JanZachmann**
return format:
```
"provisioning-config":
{
  "source": string("dps" | "manual")
  "method": string("tpm", "x509", "sas", "symmetric_key"
  // optional, only if "method": "x509"
  "x509"?: 
  {
    est: bool
    expires: string
  }
}
```

### CODE REVIEW

File: src/system.rs [hunk 1]
```diff
@@ -12,32 +12,37 @@ static DEV_OMNECT: &str = "/dev/omnect/";
 static NETWORK_SERVICE: &str = "systemd-networkd.service";
 static NETWORK_SERVICE_RELOAD_TIMEOUT_IN_SECS: u64 = 30;
 
-fn info_current_root() -> Result<()> {
+macro_rules! identity_config_file_path {
+    () => {{
+        const ENV_FILE_PATH_DEFAULT: &'static str = "/mnt/factory/etc/aziot/config.toml";
```
Comment (mlilien) [inline src/system.rs:17]: that's not necessarily the currently used config file. 
`/etc/aziot/config.toml` is the current used one and can potentially differ.

Comment (JanZachmann) [reply src/system.rs:17]: copy paste mistake

File: src/system.rs [hunk 2]
```diff
@@ -14,11 +15,18 @@ static NETWORK_SERVICE_RELOAD_TIMEOUT_IN_SECS: u64 = 30;
...
     () => {{
-        const ENV_FILE_PATH_DEFAULT: &'static str = "/mnt/factory/etc/aziot/config.toml";
+        const ENV_FILE_PATH_DEFAULT: &'static str = "/etc/aziot/config.toml";
         std::env::var("IDENTITY_CONFIG_FILE_PATH").unwrap_or(ENV_FILE_PATH_DEFAULT.to_string())
     }};
 }
 
+macro_rules! device_cert_file_path {
+    () => {{
+        const ENV_FILE_PATH_DEFAULT: &'static str = "/cert/device_id_cert.pem";
```
Comment (mlilien) [inline src/system.rs:25]: /var/lib/aziot/certd/certs/deviceid-*.cer
* is variable

File: src/system.rs [hunk 3]
```diff
@@ -70,3 +83,144 @@ pub fn sw_version() -> Result<serde_json::Value> {
...
+            "x509": {
+                "expires": not_after()?,
+                "est": false
+            }
+        })),
+        _ => bail!("invalid provisioning configuration found"),
+    }
+}
+
+fn not_after() -> Result<String> {
```
Comment (mlilien) [inline src/system.rs:152]: `systemd::not_after()` the method name doesn't reflect certificate context.

**Review by mlilien — APPROVED** [review]

**Pull request merged**